src/entities/pools/index.ts

```diff
@@ -33,3 +33,39 @@ export interface BasePoolFactory {
     isPoolForFactory(pool: RawPool): boolean;
     create(chainId: number, pool: RawPool): BasePool;
 }
+
+// Returned from API and used as input
+export type PoolState = {
+    id: string;
```

Comment:
We will likely move away from poolId in the future. So I wonder if it makes sense/is practical to start using pool address instead? That way we can keep a single code base or at least a consistent interface.

Reply:
I thought about that, but `queryJoin` requires `poolId`, so we'll need it for now.
What we could do is add `address` to PoolState and use `id` only when it's strictly necessary in favor or `address` everywhere else.
I'll go ahead with that change for now, but feel free to suggest a different approach.

Reply:
At this stage that sounds good. I can't remember how poolId is actually calculated but I thought there might be a helper to do address>poolId. Another thing to check with Daniel to see what he thinks about that kind of direction in general.

Reply:
We can always query `getPoolId` onChain, but if we do go down that path, I feel like it makes more sense to do it on the API level.